Add documentation for OTel top-level spans identification (#22277)
DataDog/documentation: content/en/opentelemetry/integrations/trace_metrics.md

## Data collected

[Trace Metrics][2] are generated for service entry spans and measured spans. These span conventions are unique to Datadog, so OpenTelemetry spans are identified with the following mapping:
| OpenTelemetry Convention | Datadog Convention |
| --- | --- |
| Root span | Service entry span |
| Server span (`span.kind: server`) | Service entry span |
| Consumer span (`span.kind: consumer`) | Service entry span |
| Client span (`span.kind: client`) | Measured span |
| Producer span (`span.kind: producer`) | Measured span |
| Internal span (`span.kind: internal`) | No trace metrics generated |

> [Span kind][3] is typically set when a span is created, but can also be updated by using the [transform processor][4] to control the mapping above. For example, if trace metrics are desired for an internal span, the following configuration transforms an internal span with `http.path: "/health"` into a server span:

Reviewer: ```suggestion
[`SpanKind`][3] is typically set when a span is created, but can also be updated by using the [transform processor][4] to control the mapping above. For example, if trace metrics are desired for an internal span, the following configuration transforms an internal span with `http.path: "/health"` into a server span:
```

> [`SpanKind`][3] is typically set when a span is created, but can also be updated by using the [transform processor][4] to control the mapping above. For example, if trace metrics are desired for an internal span, the following configuration transforms an internal span with `http.path: "/health"` into a server span:

Reviewer 2: nit: this example actually marks the internal span as service entry span, not just span with trace metrics

Author: Just updated to transform into client span instead

```yaml
  transform:
    trace_statements:
      - context: span
        statements:
          - set(kind.string, "Server") where kind.string == "Internal" and attributes["http.path"] == "/health"
```

<div class="alert alert-info">

> This service entry span logic is a new behavior that may increase the number of spans that generate trace metrics. This change can be disabled if needed by enabling the <code>disable_otlp_compute_top_level_by_span_kind</code> APM feature, but this may result in OpenTelemetry spans being misidentified as service entry spans. <code>apm_config.compute_stats_by_span_kind</code> also needs to be disabled to disable computing stats by span kind for OTel traces.

Reviewer: ```suggestion
This service entry span logic may increase the number of spans that generate trace metrics. If needed, you can disable this change by enabling the <code>disable_otlp_compute_top_level_by_span_kind</code> APM feature, but this may result in OpenTelemetry spans being misidentified as service entry spans. <code>apm_config.compute_stats_by_span_kind</code> also needs to be disabled to turn off computing stats by `SpanKind` for OTel traces.
```
